src/js/Controllers/UIController.js

```diff
@@ -295,6 +297,11 @@ class UIController {
         this.listener.send(RpcFactory.UIPerformInteractionResponse(choiceID, appID, msgID))
     }
     onSystemContext(context, appID) {
+        if (context === 'MAIN' || context === 'ALERT') {
+            store.dispatch(appStoreShowWebView());
```

Comment:
Should you check that the current route is the appstore/webview before dispatching these actions?

Reply:
You're right, there's an issue here if an app registers as WEB_VIEW and changes templates. That app's iframe would still be shown. I will look into this.

Reply:
In https://github.com/smartdevicelink/generic_hmi/pull/291/commits/a9d1f48c833287bdebb4d097b5e5794e06a7cb57 I added a variable to the state to check whether we were in `/web-view` and used this in index.js alongside this variable that reflected the current context. This worked fine, but I realized that checking if we are in `/web-view` is enough and the context doesn't matter after that. So in https://github.com/smartdevicelink/generic_hmi/pull/291/commits/a1ab7d83384407b8e2808c4727fc9ccfe0bba6e2 I removed the old state variable related to the context.